helix-term/src/commands.rs

```diff
@@ -1176,25 +1176,55 @@ fn goto_file_impl(cx: &mut Context, action: Action) {
     let primary = selections.primary();
     // Checks whether there is only one selection with a width of 1
     if selections.len() == 1 && primary.len() == 1 {
-        let count = cx.count();
-        let text_slice = text.slice(..);
-        // In this case it selects the WORD under the cursor
-        let current_word = textobject::textobject_word(
-            text_slice,
-            primary,
-            textobject::TextObject::Inside,
-            count,
-            true,
-        );
-        // Trims some surrounding chars so that the actual file is opened.
-        let surrounding_chars: &[_] = &['\'', '"', '(', ')'];
         paths.clear();
-        paths.push(
-            current_word
-                .fragment(text_slice)
-                .trim_matches(surrounding_chars)
-                .to_string(),
-        );
+
+        let is_valid_path_char = |c: &char| {
+            let valid_chars: &[char] = if cfg!(target_os = "windows") {
+                &[
+                    '@', '/', '\\', '.', '-', '_', '+', '#', '$', '%', '{', '}', '[', ']', ':',
+                    '!', '~', '=',
+                ]
+            } else {
+                &['@', '/', '.', '-', '_', '+', '#', '$', '%', '~', '=']
+            };
+            valid_chars.contains(c) || c.is_alphabetic()
+        };
+
+        let path = {
+            let start = {
```

Comment:
The logic here is unnecessarily complicated. Simply start counting (see other comment) at the next/prev char (that will be required to address the other comment anyway)

Reply:
I hope that it's better now.